bottlecap/src/logs/processor.rs

```diff
@@ -0,0 +1,120 @@
+use std::error::Error;
+
+use serde::Serialize;
+
+use crate::telemetry::events::{TelemetryEvent, TelemetryRecord};
+
+const LOGS_SOURCE: &str = "lambda";
+
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct Lambda {
+    pub arn: String,
+    pub request_id: String,
+}
+
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct LambdaMessage {
+    pub message: String,
+    pub lambda: Lambda,
+    pub timestamp: i64,
+    pub status: String,
+}
+
+#[derive(Serialize, Debug, Clone, PartialEq)]
+pub struct IntakeLog {
+    pub message: LambdaMessage,
+    pub hostname: String,
+    pub service: String,
+    #[serde(rename(serialize = "ddtags"))]
+    pub tags: String,
+    #[serde(rename(serialize = "ddsource"))]
+    pub source: String,
+}
+
+#[derive(Clone)]
+pub struct Processor {
+    function_arn: String,
+}
+
+impl Processor {
+    pub fn new(function_arn: String) -> Processor {
+        Processor { function_arn }
+    }
+
+    pub fn process(&self, event: TelemetryEvent) -> Result<IntakeLog, Box<dyn Error>> {
```

Comment:
Could you have unit tests around this?

Reply:
Yes, there's not that many unit tests yet

Reply:
LMK if I should add more!